### ISSUE TITLE
Support pure Java solver for constraint-based modeling

### ISSUE THREAD

**Issue by draeger** [draeger-lab/SBSCL#70]
On some platforms, it might be difficult to run the Java Native Interface (JNI) needed for SCPsolver. Consider adding support for pure Java implementations, such as
* https://www.ojalgo.org/code-examples/
* https://advancedweb.hu/running-lp-and-ip-optimisation-from-java/

**Comment by dyrpsf** [comment]
Hi all,

I’ve implemented a pure Java LP solver backend based on **ojAlgo**, so constraint-based modeling can run without relying on the native GLPK/LPSOLVE packs.

Changes are in PR #100 and include:

- Added `org.ojalgo:ojalgo` as a dependency in `pom.xml`.
- Introduced `org.simulator.fba.OjAlgoLinearProgramSolver`, which implements `scpsolver.lpsolver.LinearProgramSolver` and internally maps SCPSolver’s `LinearProgram` (objective, bounds, linear constraints, integrality) to ojAlgo’s `ExpressionsBasedModel`.
- Added a JUnit 5 test `OjAlgoLinearProgramSolverTest` (package `org.sbscl.fba`) that verifies the new solver on a small LP example.

Usage example:

```java
import org.simulator.fba.FluxBalanceAnalysis;
import org.simulator.fba.OjAlgoLinearProgramSolver;
import org.sbml.jsbml.SBMLDocument;

// ...
SBMLDocument doc = ...;
FluxBalanceAnalysis fba = new FluxBalanceAnalysis(doc, new OjAlgoLinearProgramSolver());
boolean ok = fba.solve();
double objective = fba.getObjectiveValue();
```

All tests (mvn test) pass on my setup (Windows 11, JDK 24, Maven 3.9.x).
Existing SCPSolver + GLPK integration is unchanged; this just provides an additional pure Java backend for environments where JNI-based solvers are hard to deploy.

Happy to adjust naming, API, or tests if you’d like it structured differently.

**Comment by draeger** [comment]
My suggestion to move forward with this issue is to focus on reviewing @xts-Michi's pull requests first. Next, I'd like to check with @dyrpsf and @xts-Michi what parts of #100 are still current. We'll need a second review for the PRs. It would be great if @matthiaskoenig could take a look as well.

**Comment by dyrpsf** [comment]
> My suggestion to move forward with this issue is to focus on reviewing [@xts-Michi](https://github.com/xts-Michi)'s pull requests first. Next, I'd like to check with [@dyrpsf](https://github.com/dyrpsf) and [@xts-Michi](https://github.com/xts-Michi) what parts of [#100](https://github.com/draeger-lab/SBSCL/pull/100) are still current. We'll need a second review for the PRs. It would be great if [@matthiaskoenig](https://github.com/matthiaskoenig) could take a look as well.

From my side, the work in OptSolvX (ojAlgo backend) and its integration into SBSCL via PRs #88 and #90 already covers what I was aiming for with PR #100.

Given that, I don’t think there is anything from PR #100 that still needs to be merged into SBSCL directly. I’m completely fine with that PR remaining closed and with this issue being resolved once the OptSolvX–SBSCL integration is finished.

After that integration has settled, I’m still happy to help with any follow‑up tasks (tests, examples, docs, or small changes on the SBSCL side). Please feel free to @‑mention me where contributions are welcome.
